docs/guides/migrating-to-ev-abci.md

```diff
@@ -0,0 +1,257 @@
+	app.UpgradeKeeper.SetUpgradeHandler(
+		"v2-migrate-to-evolve", // Upgrade name must match governance proposal
+		func(ctx sdk.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+			// The upgrade handler can initialize state for the migration manager if needed
+			// The actual migration will happen during the evolve-migrate step
+			return app.mm.RunMigrations(ctx, app.configurator, fromVM)
+		},
+	)
+}
+```
+
+Call this function in your app initialization code in `app.go`.
+
+---
+
+## Phase 3: Create Governance Proposal for Migration
+
+Create and submit a software upgrade governance proposal to initiate the migration at a specific block height.
+
+```bash
+# Create the governance proposal
+<appd> tx gov submit-proposal software-upgrade v2-migrate-to-evolve \
+  --title "Migrate to Evolve" \
+  --description "Upgrade chain to use ev-abci consensus" \
+  --upgrade-height <target-height> \
+  --from <your-key> \
+  --chain-id <chain-id>
+
+# Vote on the proposal (repeat for validators to reach quorum)
+<appd> tx gov vote <proposal-id> yes --from <validator-key>
+```
+
+Wait for the proposal to pass and for the chain to reach the upgrade height. The chain will halt at the specified height, waiting for the upgrade to be applied.
+
+---
+
+## Phase 4: Wire ev-abci Start Handler in root.go
+
+After the governance proposal passes, modify your node's entrypoint to use the `ev-abci` server commands.
+
+### Locate Your Application's Entrypoint
+
+Open the main entrypoint file for your chain's binary, usually found at `cmd/<your-app-name>/main.go` or `root.go`.
+
+### Modify the Start Command
+
+Add the `ev-abci` start handler to your root command. This is similar to the [Ignite Apps evolve template](https://github.com/ignite/apps/blob/main/evolve/template/init.go#L48-L60).
+
+```go
+// cmd/<appd>/main.go (or root.go)
+package main
+
+import (
+	"os"
+
+	"github.com/cosmos/cosmos-sdk/server"
+	"github.com/spf13/cobra"
+
+	// Import the ev-abci server package
+	evabci_server "github.com/evstack/ev-abci/server"
+
+	"<your-app-path>/app"
+)
+
+func main() {
+	rootCmd := &cobra.Command{
+		Use:   "<appd>",
+		Short: "Your App Daemon (ev-abci enabled)",
+	}
+
+	// Keep existing commands (keys, export, etc.)
+	server.AddCommands(rootCmd, app.DefaultNodeHome, app.New, app.MakeEncodingConfig(), tx.DefaultSignModes)
+
+	// --- Wire ev-abci start handler ---
+	startCmd := &cobra.Command{
+		Use:   "start",
+		Short: "Run the full node with ev-abci",
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return server.Start(cmd, evabci_server.StartHandler())
+		},
+	}
+
+	evabci_server.AddFlags(startCmd)
+	rootCmd.AddCommand(startCmd)
+	// --- End of ev-abci changes ---
+
+	if err := rootCmd.Execute(); err != nil {
+		server.HandleError(err)
+		os.Exit(1)
+	}
+}
+```
+
+### Build Your Application
+
+Re-build your application's binary with the updated code:
+
+```sh
+go build -o <appd> ./cmd/<appd>
+```
+
```

Comment:
Let's add a disclaimer that the node should NOT be started. Otherwise we can just move phase 4 after phase 5.